metron-analytics/metron-profiler-client/src/test/java/org/apache/metron/profiler/client/GetProfileTest.java

```diff
@@ -116,6 +119,33 @@ public void setup() {
                     .build());
   }
 
+  private Context setup2() {
```

Comment:
This is a test to demonstrate that the "current" configuration can be set differently than the "historical" configuration, and then both the positive and negative test cases work right.  `setup2()` is modeled exactly on `setup()`, except that it uses `periodDuration2`, `periodUnits2` and `saltDivisor2`, instead of `periodDuration`, `periodUnits` and `saltDivisor` respectively.

However, you're right that this is a little opaque, because setup() is called in the @Before clause, while setup2() is called in the midst of the test routine.  Let me add some comments that clarify this.